plugins/processors/nginx_vts_filter/nginx_vts_filter.go

```diff
@@ -0,0 +1,170 @@
+			//     backend = "",
+			//     status = "401",
+			//     filter_name = "client=Mesos/1.8.0 authorizer (master)",
+			// }
+			filter, ok := point.GetTag("filter")
+			if !ok {
+				log.Printf("E! [processors.nginx_vts_filter] %s has no tag filter", name)
+				continue
+			}
+			filterTags, err := unwrapTags(filter)
+			if err != nil {
+				log.Printf("E! [processors.nginx_vts_filter] could not unwrap filter tags %s: %v", filter, err)
+				continue
+			}
+			for _, t := range filterTags {
+				point.AddTag(t.Key, t.Value)
+			}
+			point.RemoveTag("filter")
+			//
+			// Step 2: Convert filter_name tag value into new tag.
+			// ------------------------------------------------------------
+			// nginx_vts_filter_requests_total{
+			//     upstream = "Bouncer",
+			//     backend = "",
+			//     status = "401",
+			//     filter_name = "client=Mesos/1.8.0 authorizer (master)",
+			// }
+			// ------------------------------------------------------------
+			// nginx_vts_filter_requests_total{
+			//     upstream = "Bouncer",
+			//     backend = "",
+			//     status = "401",
+			//     client = "Mesos/1.8.0 authorizer (master)",
+			// }
+			filterName, ok := point.GetTag("filter_name")
+			if !ok {
+				log.Printf("E! [processors.nginx_vts_filter] %s has no tag filter_name", name)
+				continue
+			}
+			filterNameTags, err := unwrapTags(filterName)
+			if err != nil {
+				log.Printf("E! [processors.nginx_vts_filter] could not unwrap filter_name tags %s: %v", filterName, err)
+				continue
+			}
+			for _, t := range filterNameTags {
+				point.AddTag(t.Key, t.Value)
+			}
+			point.RemoveTag("filter_name")
+			//
+			// Step 3: Convert metric name via filter/filter_name tag keys.
+			// ------------------------------------------------------------
+			// nginx_vts_filter_requests_total{
+			//     upstream = "Bouncer",
+			//     backend = "",
+			//     status = "401",
+			//     client = "Mesos/1.8.0 authorizer (master)",
+			// }
+			// ------------------------------------------------------------
+			// nginx_upstream_client_requests_total{
+			//     upstream = "Bouncer",
+			//     backend = "",
+			//     status = "401",
+			//     client = "Mesos/1.8.0 authorizer (master)",
+			// }
+			newName, err := vtsFilterName(name, filterTags, filterNameTags)
+			if err != nil {
+				log.Printf("E! [processors.nginx_vts_filter] could not rename %s: %v", name, err)
+				continue
+			}
+			point.SetName(newName)
+		}
+	}
+	return in
+}
+
+func init() {
+	processors.Add("nginx_vts_filter", func() telegraf.Processor {
+		return &NginxVTSFilter{}
+	})
+}
+
+func kvStringToTag(kv string) (telegraf.Tag, error) {
+	tuple := strings.Split(kv, ":=")
+	if len(tuple) != 2 {
+		return telegraf.Tag{}, fmt.Errorf("not a key-value pair: %s", kv)
+	}
+	return telegraf.Tag{
+		Key:   tuple[0],
+		Value: tuple[1],
+	}, nil
+}
+
+func unwrapTags(kvPairs string) ([]telegraf.Tag, error) {
+	// Valid filter value format: "a:=x::b:=y::c:=z"
```

Comment:
If we're going to require `::` as a delimiter we should update `README.md` file. Maybe it would be possible to make it configurable?